docs/api-reference/next.config.js/rewrites.md

```diff
@@ -318,6 +323,26 @@ module.exports = {
 }
 ```
 
+If you're using `trailingSlash: true`, you also need to insert a trailing slash in the rewrites and use a special wildcard.
+
+```js
+module.exports = {
+  trailingSlash: 'true',
+  async rewrites() {
+    return [
+      {
+        source: '/blog/',
+        destination: 'https://example.com/blog/',
+      },
+      {
+        source: '/blog/:path(.+)',
```

Comment:
This seems like a bug, wdyt @ijjk 🤔 
```suggestion
        source: '/blog/:path(.+)',
```

Reply:
For context, I discovered that `:path*` doesn't catch trailing slashes so Davis came up with a wildcard expression that does.

Reply:
We should be handling the trailing slash here https://github.com/vercel/next.js/blob/1a0d149c9ff6afa061b902c32b51aa848141cefd/packages/next/lib/load-custom-routes.ts#L72-L73 can you provide a reproduction where this isn't working?

Reply:
@ijjk yep! Here's a reproduction. Click on the "not working" demo link below, navigate to a specific blog post, then reload the page. It will redirect infinitely.

- [Not working demo of main repo with `:path*`](https://redirect-demo-git-not-working-nutlope.vercel.app/blog/)
- [Working demo of main repo with `:path(.+)`](https://redirect-demo.vercel.app/blog/)

If you want to take a look at the code, here are the full repos. They both have `trailingSlash: true`:

- [not working main repo](https://github.com/Nutlope/redirect-demo/tree/not-working) 
- [working main repo](https://github.com/Nutlope/redirect-demo)
- [blog repo](https://github.com/Nutlope/starter-blog-example)

Reply:
Oh it looks like the `destination` for [this rewrite](https://github.com/Nutlope/redirect-demo/blob/b018414bbbc9cc5cb3c28ee5b0670327759f6a2f/next.config.js#L53) needs to end with a trailingSlash to match the other deployments expectation e.g. does changing it to the below resolve this?

```
destination: `${BLOG_URL}/blog/:path*/`
```

Reply:
I think documenting that the `destination` for a rewrite needs to honor the external destination's trailingSlash config makes sense

Reply:
Good catch, it works when you include a trailing slash for both the source and destination like this. I'll update the PR to reflect this instead of the `:path(.+)` wildcard

```js
{
  source: "/blog/:path*/",
  destination: `${BLOG_URL}/blog/:path*/`,
},
```